### PR TITLE
Add druid.indexer.server.maxChatRequests for QoS; deprecate separate ports.

### PR DESCRIPTION
**Pull request by gianm**
- Add druid.indexer.server.maxChatRequests, which sets up a QoSFilter on the main Jetty server.
- Deprecate druid.indexer.runner.separateIngestionEndpoint
- Deprecate druid.indexer.server.chathandler.*

### CODE REVIEW

**Comment by gianm** [comment]
see discussion on #2419

**Comment by fjy** [comment]
@gianm 

Running io.druid.server.initialization.JettyQosTest
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 53.158 sec <<< FAILURE! - in io.druid.server.initialization.JettyQosTest
testQoS(io.druid.server.initialization.JettyQosTest)  Time elapsed: 52.729 sec  <<< FAILURE!
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at io.druid.server.initialization.JettyQosTest.testQoS(JettyQosTest.java:191)

**Comment by fjy** [comment]
@nishantmonu51 can you take a look?

**Comment by gianm** [comment]
@fjy fixed the test

File: server/src/test/java/io/druid/server/initialization/JettyQosTest.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                  slowClient.go(
+                      new Request(HttpMethod.GET, new URL("http://localhost:" + port + "/slow/hello")),
+                      new StatusResponseHandler(Charset.defaultCharset())
+                  );
+              go.get();
+              slowCount.incrementAndGet();
+              slowElapsed.addAndGet(System.currentTimeMillis() - startTime);
+            }
+            catch (InterruptedException e) {
+              // BE COOL
```
Comment (fjy) [inline server/src/test/java/io/druid/server/initialization/JettyQosTest.java]: ?

**Comment by fjy** [comment]
:+1:

**Comment by gianm** [comment]
would like feedback from at least a couple of @nishantmonu51 @drcrallen @guobingkun @pjain1 or @cheddar on this one, as it is a follow up to https://groups.google.com/forum/#!msg/druid-development/hfCggehFl_I/INfjEyxbBQAJ and deprecates the feature introduced in #1929.

**Comment by nishantmonu51** [comment]
looks good to me, :+1: from my side, still keeping it open for others to review it.

**Comment by drcrallen** [comment]
Suggest leaving open and bringing up in the dev sync to see if anyone wants to comment (also stating why this PR is needed). Then merge if no comment.

**Comment by cheddar** [comment]
I didn't look too closely, but the goal of #1929 was to provide for clear separation of ingestion and query resources.  As long as this maintains that (which it looks like it does), I'm on board.

**Comment by pjain1** [comment]
:+1:

File: server/src/main/java/io/druid/server/initialization/jetty/ChatHandlerServerModule.java [hunk 2]
```diff
@@ -50,15 +54,28 @@
...
   public void configure(Binder binder)
   {
-    /** If "druid.indexer.task.chathandler.port" property is set then we assume that a
-     * separate Jetty Server with it's own {@link ServerConfig} is required for ingestion apart from the query server
-     * otherwise we bind {@link DruidNode} annotated with {@link RemoteChatHandler} to {@literal @}{@link Self} {@link DruidNode}
-     * so that same Jetty Server is used for querying as well as ingestion
+    Jerseys.addResource(binder, ChatHandlerResource.class);
+
+    if (properties.containsKey(MAX_CHAT_REQUESTS_PROPERTY)) {
+      final int maxRequests = Integer.parseInt(MAX_CHAT_REQUESTS_PROPERTY);
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/initialization/jetty/ChatHandlerServerModule.java]: Can't believe I missed this... I'm pretty sure this is NOT what you wanted to do. Also this needs a parse catcher

Comment (gianm) [inline server/src/main/java/io/druid/server/initialization/jetty/ChatHandlerServerModule.java]: really good point

Comment (gianm) [inline server/src/main/java/io/druid/server/initialization/jetty/ChatHandlerServerModule.java]: does it need a parse catcher? I think it's ok for this to blow up and let Guice report the error if you pass something that isn't a number

**Pull request merged**